cmd/ops_agent_uap_plugin/service_windows.go

```diff
@@ -53,9 +195,280 @@ func (ps *OpsAgentPluginServer) Stop(ctx context.Context, msg *pb.StopRequest) (
+		uintptr(unsafe.Pointer(&info)),
+		uint32(unsafe.Sizeof(info)))
+	if err != nil {
+		windows.CloseHandle(jobHandle)
+		return 0, err
+	}
+
+	return jobHandle, nil
 }
 
-func runCommand(cmd *exec.Cmd) (string, error) {
-	panic("runCommand method is not implemented on Windows yet")
+// runSubagents starts up the diagnostics service, otel, and fluent bit subagents in separate goroutines.
+// All child goroutines create a new context derived from the same parent context.
+// This ensures that crashes in one goroutine don't affect other goroutines.
+// However, when one goroutine exits with errors, it won't be restarted, and all other goroutines are also terminated.
+// This is done by canceling the parent context.
+// This makes sure that GetStatus() returns a non-healthy status, signaling UAP to Start() the plugin again.
+//
+// ctx: the parent context that all child goroutines share.
+//
+// cancel: the cancel function for the parent context. By calling this function, the parent context is canceled,
+// and GetStatus() returns a non-healthy status, signaling UAP to re-trigger Start().
+//
+// otelErrorHandler: an implementation of otel.ErrorHandler that is used in the diagnostics service to log otel errors to the Windows event log.
+//
+// jobHandle: a Windows Job object handle. This is used to ensure that all child processes are killed when the parent plugin grpc server process exits.
+func runSubagents(ctx context.Context, cancel context.CancelFunc, pluginInstallDirectory string, pluginStateDirectory string, runSubAgentCommand RunSubAgentCommandFunc, runCommand RunCommandFunc, otelErrorHandler otel.ErrorHandler, jobHandle windows.Handle) {
+
+	var wg sync.WaitGroup
+	// Starting the diagnostics service
+	wg.Add(1)
+	go runDiagnosticsService(ctx, cancel, otelErrorHandler, &wg)
+
+	// Starting Otel
+	runOtelCmd := exec.CommandContext(ctx,
+		path.Join(pluginInstallDirectory, OtelBinary),
+		"--config", path.Join(pluginStateDirectory, GeneratedConfigsOutDir, "otel/otel.yaml"),
+	)
+	wg.Add(1)
+	go runSubAgentCommand(ctx, cancel, runOtelCmd, runCommand, &wg, jobHandle)
+
+	// Starting Fluentbit
+	runFluentBitCmd := exec.CommandContext(ctx,
+		path.Join(pluginInstallDirectory, AgentWrapperBinary),
+		"-config_path", OpsAgentConfigLocationWindows,
+		"-log_path", path.Join(pluginStateDirectory, LogsDirectory, "logging-module.log"),
+		path.Join(pluginInstallDirectory, FluentbitBinary),
+		"-c", path.Join(pluginStateDirectory, GeneratedConfigsOutDir, "fluentbit/fluent_bit_main.conf"),
+		"-R", path.Join(pluginStateDirectory, GeneratedConfigsOutDir, "fluentbit/fluent_bit_parser.conf"),
+		"--storage_path", path.Join(pluginStateDirectory, "run/buffers"),
+	)
+	wg.Add(1)
+	go runSubAgentCommand(ctx, cancel, runFluentBitCmd, runCommand, &wg, jobHandle)
+
+	wg.Wait()
+	windows.CloseHandle(jobHandle)
+}
+
+func runDiagnosticsService(ctx context.Context, cancel context.CancelFunc, otelErrorHandler otel.ErrorHandler, wg *sync.WaitGroup) {
```

Comment:
Why do we not assign the diagnostics service to the JobHandle. Do we plan on manually ensuring it's killed?

Reply:
Although the runDiagnosticsService function runs in a goroutine, it does not start any os processes, unlike the otel and fluentbit goroutines.

child goroutines are terminated when the parent program terminates. This is handled by go.